bootloader/bootloader.go

```diff
@@ -291,9 +295,14 @@ func Find(rootdir string, opts *Options) (Bootloader, error) {
 		opts = &Options{}
 	}
 
-	for _, blNew := range bootloaders {
+	// note that the order of this is not deterministic
+	for name, blNew := range bootloaders {
 		bl := blNew(rootdir, opts)
-		if osutil.FileExists(bl.ConfigFile()) {
+		present, err := bl.Present()
+		if err != nil {
+			return nil, fmt.Errorf("bootloader %q found but not usable: %v", name, err)
```

Comment:
this could be bl.Name() and then we don't need bootloaders to be a map[string]bootloaderNewFunc

Reply:
this too